test/e2e/proxy_test.go

```diff
@@ -68,7 +70,493 @@ func skipIfProxyDisabled(t *testing.T) {
+	port := "8080"
+	clientCpIPv4, clientWkIPv4, clientCpIPv6, clientWkIPv6 := createTestClientPods(t, data, clientPodCp, clientPodWk)
+	clientCpIP := clientCpIPv4
+	clientWkIP := clientWkIPv4
+	if isIPv6 {
+		clusterIngressIP = []string{"fd75::aabb:ccdd:ef00"}
+		localIngressIP = []string{"fd75::aabb:ccdd:ef01"}
+		clientCpIP = clientCpIPv6
+		clientWkIP = clientWkIPv6
+	}
+
+	createTestAgnhostPods(t, data, testPodNameCp, testPodNameWk, false)
+	createLoadBalancerServices(t, data, clusterIngressIP, localIngressIP, isIPv6)
+	clusterUrl := net.JoinHostPort(clusterIngressIP[0], port)
+	localUrl := net.JoinHostPort(localIngressIP[0], port)
+
+	t.Run("Pod CIDR Endpoints", func(t *testing.T) {
+		loadBalancerTestCases(t, data, clusterUrl, localUrl, clientPodCp, clientPodWk, clientCpIP, clientWkIP, testPodNameCp, testPodNameWk)
+	})
+
+	testPodHostNetworkNameCp := "agnhost-cp-h"
+	testPodHostNetworkNameWk := "agnhost-wk-h"
+	nodeNameCp := controlPlaneNodeName()
+	nodeNameWk := workerNodeName(1)
+	deleteTestAgnhostPods(t, data, testPodNameCp, testPodNameWk)
+	createTestAgnhostPods(t, data, testPodHostNetworkNameCp, testPodHostNetworkNameWk, true)
+	t.Run("Host Network Endpoints", func(t *testing.T) {
+		loadBalancerTestCases(t, data, clusterUrl, localUrl, clientPodCp, clientPodWk, clientCpIP, clientWkIP, nodeNameCp, nodeNameWk)
+	})
+}
+
+func loadBalancerTestCases(t *testing.T, data *TestData, clusterUrl, localUrl, clientCp, clientWk, clientCpIP, clientWkIP,
+	testPodHostnameCp, testPodHostnameWk string) {
+	t.Run("Case=ExternalTrafficPolicy:Cluster Client:Local", func(t *testing.T) {
+		testLoadBalancerClusterFromLocal(t, data, clusterUrl)
+	})
+	t.Run("Case=ExternalTrafficPolicy:Cluster Client:Pod", func(t *testing.T) {
+		testLoadBalancerClusterFromPod(t, data, clusterUrl, clientCp, clientWk)
+	})
+	t.Run("Case=ExternalTrafficPolicy:Local Client:Local", func(t *testing.T) {
+		testLoadBalancerLocalFromLocal(t, data, localUrl, testPodHostnameCp, testPodHostnameWk)
+	})
+	t.Run("Case=ExternalTrafficPolicy:Local Client:Pod", func(t *testing.T) {
+		testLoadBalancerLocalFromPod(t, data, localUrl, clientCp, clientWk, clientCpIP, clientWkIP, testPodHostnameCp, testPodHostnameWk)
+	})
+}
+
+func createLoadBalancerServices(t *testing.T, data *TestData, ingressIPCluster, ingressIPLocal []string, isIPv6 bool) {
+	ipProctol := corev1.IPv4Protocol
+	if isIPv6 {
+		ipProctol = corev1.IPv6Protocol
+	}
+	_, err := data.createAgnhostLoadBalancerService("agnhost-cluster", true, false, ingressIPCluster, &ipProctol)
+	require.NoError(t, err)
+	_, err = data.createAgnhostLoadBalancerService("agnhost-local", true, true, ingressIPLocal, &ipProctol)
+	require.NoError(t, err)
+}
+
+func testLoadBalancerClusterFromLocal(t *testing.T, data *TestData, url string) {
+	errMsg := "Server LoadBalancer whose externalTrafficPolicy is Cluster should be able to be connected from local"
+
+	nodeCp := controlPlaneNodeName()
+	skipIfKubeProxyEnabledOnLinux(t, data, nodeCp)
```

Comment:
Why it checks kube-proxy for each Node, isn't this a per-cluster configuration?

Reply:
Updated.